### PR TITLE
Add new examples to help files

### PR DESCRIPTION
**Pull request by glGarg**
- Reorder commands in help files to be alphabetical
 - Add new crafted examples to help files
---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] The PR has modified HISTORY.rst describing any customer-facing, functional changes. Note that this does not include changes only to help content. (see [Modifying change log](https://github.com/Azure/azure-cli/tree/master/doc/authoring_command_modules#modify-change-log)).

- [x] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

### CODE REVIEW

File: src/command_modules/azure-cli-appservice/azure/cli/command_modules/appservice/_help.py [hunk 1]
```diff
@@ -82,11 +94,19 @@
...
+    text: az functionapp config appsettings list --name MyWebapp --resource-group MyResourceGroup
+    crafted: true
 """
 
 helps['functionapp config appsettings set'] = """
 type: command
 short-summary: Update a function app's settings.
+examples:
+  - name: Update a function app's settings. (crafted)
+    text: az functionapp config appsettings set --name MyFunctionApp --resource-group MyResourceGroup --settings {settings}
```
Comment (tjprescott) [inline src/command_modules/azure-cli-appservice/azure/cli/command_modules/appservice/_help.py]: Remove.

File: src/command_modules/azure-cli-appservice/azure/cli/command_modules/appservice/_help.py [hunk 2]
```diff
@@ -137,6 +157,10 @@
 helps['functionapp config set'] = """
 type: command
 short-summary: Set the web app's configuration.
+examples:
+  - name: Set the web app's configuration. (crafted)
+    text: az functionapp config set --min-tls-version {min-tls-version} --name MyFunctionApp --resource-group MyResourceGroup
```
Comment (tjprescott) [inline src/command_modules/azure-cli-appservice/azure/cli/command_modules/appservice/_help.py]: Remove.

File: src/command_modules/azure-cli-appservice/azure/cli/command_modules/appservice/_help.py [hunk 3]
```diff
@@ -253,6 +293,13 @@
 helps['functionapp deployment source config'] = """
 type: command
 short-summary: Manage deployment from git or Mercurial repositories.
+examples:
+  - name: Manage deployment from git or Mercurial repositories. (crafted)
+    text: az functionapp deployment source config --branch {branch} --name MyFunctionApp --repo-url {repo-url} --repository-type git --resource-group MyResourceGroup
+    crafted: true
+  - name: Perform deployment using the kudu zip push deployment for a function app. (crafted)
+    text: az functionapp deployment source config-zip --name MyFunctionApp --resource-group MyResourceGroup --src {src}
```
Comment (tjprescott) [inline src/command_modules/azure-cli-appservice/azure/cli/command_modules/appservice/_help.py]: Remove

File: src/command_modules/azure-cli-appservice/azure/cli/command_modules/appservice/_help.py [hunk 4]
```diff
@@ -453,11 +528,19 @@
 helps['webapp config appsettings delete'] = """
 type: command
 short-summary: Delete web app settings.
+examples:
+  - name: Delete web app settings. (crafted)
+    text: az webapp config appsettings delete --name MyWebapp --resource-group MyResourceGroup --setting-names {setting-names}
```
Comment (tjprescott) [inline src/command_modules/azure-cli-appservice/azure/cli/command_modules/appservice/_help.py]: Remove

File: src/command_modules/azure-cli-appservice/azure/cli/command_modules/appservice/_help.py [hunk 5]
```diff
@@ -510,11 +593,19 @@
 helps['webapp config connection-string delete'] = """
 type: command
 short-summary: Delete a web app's connection strings.
+examples:
+  - name: Delete a web app's connection strings. (crafted)
+    text: az webapp config connection-string delete --name MyWebapp --resource-group MyResourceGroup --setting-names {setting-names}
```
Comment (tjprescott) [inline src/command_modules/azure-cli-appservice/azure/cli/command_modules/appservice/_help.py]: Remove

File: src/command_modules/azure-cli-appservice/azure/cli/command_modules/appservice/_help.py [hunk 6]
```diff
@@ -540,11 +631,19 @@
 helps['webapp config container set'] = """
 type: command
 short-summary: Set a web app container's settings.
+examples:
+  - name: Set a web app container's settings. (crafted)
+    text: az webapp config container set --docker-custom-image-name MyDockerCustomImage --docker-registry-server-password {docker-registry-server-password} --docker-registry-server-url {docker-registry-server-url} --docker-registry-server-user {docker-registry-server-user} --name MyWebapp --resource-group MyResourceGroup
+    crafted: true
```
Comment (tjprescott) [inline src/command_modules/azure-cli-appservice/azure/cli/command_modules/appservice/_help.py]: Remove

File: src/command_modules/azure-cli-appservice/azure/cli/command_modules/appservice/_help.py [hunk 7]
```diff
@@ -555,6 +654,10 @@
 helps['webapp config hostname add'] = """
 type: command
 short-summary: Bind a hostname to a web app.
+examples:
+  - name: Bind a hostname to a web app. (crafted)
+    text: az webapp config hostname add --hostname {hostname} --resource-group MyResourceGroup --webapp-name MyWebapp
```
Comment (tjprescott) [inline src/command_modules/azure-cli-appservice/azure/cli/command_modules/appservice/_help.py]: Remove

File: src/command_modules/azure-cli-appservice/azure/cli/command_modules/appservice/_help.py [hunk 8]
```diff
@@ -793,6 +932,10 @@
 helps['webapp deployment slot create'] = """
 type: command
 short-summary: Create a deployment slot.
+examples:
+  - name: Create a deployment slot. (crafted)
+    text: az webapp deployment slot create --name MyWebapp --resource-group MyResourceGroup --slot {slot}
```
Comment (tjprescott) [inline src/command_modules/azure-cli-appservice/azure/cli/command_modules/appservice/_help.py]: Remove

File: src/command_modules/azure-cli-appservice/azure/cli/command_modules/appservice/_help.py [hunk 9]
```diff
@@ -823,6 +970,13 @@
 helps['webapp deployment source config'] = """
 type: command
 short-summary: Manage deployment from git or Mercurial repositories.
+examples:
+  - name: Get a URL for a git repository endpoint to clone and push to for web app deployment. (crafted)
+    text: az webapp deployment source config-local-git --name MyWebapp --resource-group MyResourceGroup
+    crafted: true
+  - name: Perform deployment using the kudu zip push deployment for a webapp. (crafted)
+    text: az webapp deployment source config-zip --name MyWebapp --resource-group MyResourceGroup --src {src}
```
Comment (tjprescott) [inline src/command_modules/azure-cli-appservice/azure/cli/command_modules/appservice/_help.py]: Remove

File: src/command_modules/azure-cli-appservice/azure/cli/command_modules/appservice/_help.py [hunk 10]
```diff
@@ -864,11 +1018,19 @@
 helps['webapp deployment source show'] = """
 type: command
 short-summary: Get the details of a source control deployment configuration.
+examples:
+  - name: Get the details of a source control deployment configuration. (crafted)
+    text: az webapp deployment source show --name MyWebapp --resource-group MyResourceGroup --slot {slot}
```
Comment (tjprescott) [inline src/command_modules/azure-cli-appservice/azure/cli/command_modules/appservice/_help.py]: Remove

File: src/command_modules/azure-cli-appservice/azure/cli/command_modules/appservice/_help.py [hunk 11]
```diff
@@ -938,12 +1104,20 @@
...
+    crafted: true
 """
 
 helps['webapp log download'] = """
 type: command
 short-summary: Download a web app's log history as a zip file.
 long-summary: This command may not work with web apps running on Linux.
+examples:
+  - name: Download a web app's log history as a zip file. (crafted)
+    text: az webapp log download --log-file {log-file} --name MyWebapp --resource-group MyResourceGroup --slot {slot}
```
Comment (tjprescott) [inline src/command_modules/azure-cli-appservice/azure/cli/command_modules/appservice/_help.py]: Remove

File: src/command_modules/azure-cli-dla/azure/cli/command_modules/dla/_help.py [hunk 12]
```diff
@@ -401,6 +401,10 @@
 helps['dla job show'] = """
 type: command
 short-summary: Get information for a Data Lake Analytics job.
+examples:
+  - name: Get information for a Data Lake Analytics job. (commonly used with --output)
+    text: az dla job show --account {account} --job-identity {job-identity}
```
Comment (tjprescott) [inline src/command_modules/azure-cli-dla/azure/cli/command_modules/dla/_help.py]: Remove

File: src/command_modules/azure-cli-dla/azure/cli/command_modules/dla/_help.py [hunk 13]
```diff
@@ -423,6 +427,10 @@
     short-summary: The priority of the job.
     long-summary: Lower values increase the priority, with the lowest value being 1. This determines the order jobs are run in.
 
+examples:
+  - name: Submit a job to a Data Lake Analytics account. (commonly used with --output)
+    text: az dla job submit --account {account} --degree-of-parallelism {degree-of-parallelism} --job-name MyJob   --script {script}
```
Comment (tjprescott) [inline src/command_modules/azure-cli-dla/azure/cli/command_modules/dla/_help.py]: Remove

File: src/command_modules/azure-cli-dls/azure/cli/command_modules/dls/_help.py [hunk 14]
```diff
@@ -170,6 +174,10 @@
 helps['dls fs access set-entry'] = """
 type: command
 short-summary: Update the access control list for a file or folder.
+examples:
+  - name: Update the access control list for a file or folder. (crafted)
+    text: az dls fs access set-entry --account {account} --acl-spec user:6360e05d-c381-4275-a932-5535806bb323:-w- --path {path}
```
Comment (tjprescott) [inline src/command_modules/azure-cli-dls/azure/cli/command_modules/dls/_help.py]: Remove. Concerns about the specificity of the GUID under user.

File: src/command_modules/azure-cli-resource/azure/cli/command_modules/resource/_help.py [hunk 15]
```diff
@@ -271,11 +275,19 @@
 helps['feature list'] = """
 type: command
 short-summary: List preview features.
+examples:
+  - name: List preview features. (crafted)
+    text: az feature list --namespace {namespace}
```
Comment (tjprescott) [inline src/command_modules/azure-cli-resource/azure/cli/command_modules/resource/_help.py]: Remove

File: src/command_modules/azure-cli-resource/azure/cli/command_modules/resource/_help.py [hunk 16]
```diff
@@ -271,11 +275,19 @@
...
+    crafted: true
 """
 
 helps['feature register'] = """
 type: command
 short-summary: register a preview feature.
+examples:
+  - name: register a preview feature. (crafted)
+    text: az feature register --name MyFeature --namespace {namespace}
```
Comment (tjprescott) [inline src/command_modules/azure-cli-resource/azure/cli/command_modules/resource/_help.py]: Remove

File: src/command_modules/azure-cli-resource/azure/cli/command_modules/resource/_help.py [hunk 17]
```diff
@@ -351,6 +363,10 @@
     long-summary: >
         Parameters may be supplied from a file using the `@{path}` syntax, a JSON string, or as <KEY=VALUE> pairs. Parameters are evaluated in order, so when a value is assigned twice, the latter value will be used.
         It is recommended that you supply your parameters file first, and then override selectively using KEY=VALUE syntax.
+examples:
+  - name: Validate whether a template is syntactically correct. (crafted)
+    text: az group deployment validate --parameters {parameters} --resource-group MyResourceGroup --template-file {template-file}
+    crafted: true
 """
```
Comment (tjprescott) [inline src/command_modules/azure-cli-resource/azure/cli/command_modules/resource/_help.py]: Remove

File: src/command_modules/azure-cli-resource/azure/cli/command_modules/resource/_help.py [hunk 18]
```diff
@@ -429,11 +445,19 @@
 helps['group update'] = """
 type: command
 short-summary: Update a resource group.
+examples:
+  - name: Update a resource group. (crafted)
+    text: az group update --resource-group MyResourceGroup --set {set}
```
Comment (tjprescott) [inline src/command_modules/azure-cli-resource/azure/cli/command_modules/resource/_help.py]: Remove

File: src/command_modules/azure-cli-resource/azure/cli/command_modules/resource/_help.py [hunk 19]
```diff
@@ -833,6 +861,10 @@
 helps['provider register'] = """
 type: command
 short-summary: Register a provider.
+examples:
+  - name: Register a provider. (crafted)
+    text: az provider register --namespace {namespace}
```
Comment (tjprescott) [inline src/command_modules/azure-cli-resource/azure/cli/command_modules/resource/_help.py]: Remove

File: src/command_modules/azure-cli-resource/azure/cli/command_modules/resource/_help.py [hunk 20]
```diff
@@ -1076,6 +1108,10 @@
 helps['resource update'] = """
 type: command
 short-summary: Update a resource.
+examples:
+  - name: Update a resource. (crafted)
+    text: az resource update --api-version {api-version} --ids {ids} --set {set}
```
Comment (tjprescott) [inline src/command_modules/azure-cli-resource/azure/cli/command_modules/resource/_help.py]: Remove

**Comment by tjprescott** [comment]
@marstr retargeting this for the release branch.

File: src/command_modules/azure-cli-resource/azure/cli/command_modules/resource/_help.py (hunk 20 above)
Comment (glGarg) [reply src/command_modules/azure-cli-resource/azure/cli/command_modules/resource/_help.py]: Done

**Pull request merged**